api/agent/call.go

```diff
@@ -344,14 +350,21 @@ func (c *call) End(ctx context.Context, err error) {
 	// TODO: this should be update, really
 	if err := c.ds.InsertCall(ctx, c.Call); err != nil {
 		common.Logger(ctx).WithError(err).Error("error inserting call into datastore")
+		return err
```

Comment:
we need to explicitly not do this, so that the error sent into `End` is the one returned (generally), this is for timeouts and cancellations, primarily, but also, the task could have ran successfully and been a sync task, and we don't want to return a `500` simply because we couldn't store it in the db. We also want to try to store the logs, even if storing the call failed. I removed the comment lines about this, should probably add them back since it's not obvious.